lock-impl/src/main/java/com/palantir/lock/impl/LockThreadInfoSnapshotManager.java

```diff
@@ -36,39 +41,78 @@
 import java.util.concurrent.TimeUnit;
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
+import javax.annotation.concurrent.GuardedBy;
 
 public class LockThreadInfoSnapshotManager implements AutoCloseable {
-    private DebugThreadInfoConfiguration threadInfoConfiguration;
+    private static final SafeLogger log = SafeLoggerFactory.get(LockThreadInfoSnapshotManager.class);
+    private Refreshable<DebugThreadInfoConfiguration> threadInfoConfiguration;
 
     private Supplier<ConcurrentMap<HeldLocksToken, HeldLocks<HeldLocksToken>>> tokenMapSupplier;
 
-    private volatile Map<LockDescriptor, LockClientAndThread> lastKnownThreadInfoSnapshot;
+    private volatile Map<LockDescriptor, LockClientAndThread> lastKnownThreadInfoSnapshot = ImmutableMap.of();
 
     private ScheduledExecutorService scheduledExecutorService = PTExecutors.newSingleThreadScheduledExecutor();
 
+    private Optional<Disposable> disposable;
```

Comment:
Disposable should be optional since we may never call start() before calling close()